### PR TITLE
Milestone2

### PR DESCRIPTION
**Pull request by Israel-Laguan**
# Resume
We've followed the specifications as described in the project requirements, added the redux store, required components and containers , and *came to the conclusion that we met all of the requirements*.
# Proposal
We want to have a review of the code to know if we need to make any changes. If no changes needed kindly ACCEPT this milestone so we can move to the next milestone with a happy smile :)
# Specifications
- Followed [these requirements](https://github.com/microverseinc/project-redux-bookstore/blob/master/milestones/MILESTONE_2.md) for Milestone 2
- Added code for redux
- added components
- Served from Heroku
# Live Version Here
https://bookstorecms.herokuapp.com/

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,23 +1,38 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import BooksList from './containers/BooksList';
+import BookForm from './containers/BookForm';
 
-function App() {
+
+function App({ books = [] }) {
```
Comment (bellom) [inline src/App.js:7]: Opps! we miss some hard requirement:

You need to create `component` dir and move all presentational components inside it, that's; both `app.js` and `book.js`.
Components that connect to redux store are held in `src/container` while components that don't connect to redux store are held in `src/component` and they are called presentational component.

image

image

File: src/components/Book.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
```
Comment (bellom) [inline src/components/Book.js:3]: According to pathwright `book.js` should not connect to `redux store`, the screenshot below;
Kindly follow the instructions as other students are also following it 👍 

image

File: src/components/Book.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+/**
+ * Book
+ *
+ * Book component shows the information regarding
+ * a book
+ * @param {Object} props Data from parent
+ * @return {React.Component} Book
+ */
+const Book = props => {
```
Comment (bellom) [inline src/components/Book.js:14]: Let's keep our `book` component simple, screenshot from pathwright below;

image

File: src/containers/BookForm.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    const formData = {
+      category: e.target.category.value,
+      title: e.target.title.value,
+      author: e.target.author.value,
+      pages: e.target.pages.value,
+      progress: e.target.progress.value,
+      summary: e.target.summary.value,
+    };
+    // eslint-disable-next-line no-console
+    console.log(formData);
```
Comment (bellom) [inline src/containers/BookForm.js:45]: This should be a production code, please remove debugging statements. (`console.log`), even if that feature is not fully implemented, users should not see debugging messages, then you can remove unnecessarily `eslint-disable`

File: src/App.js (hunk 1 above)
Comment (Israel-Laguan) [reply src/App.js]: Solved!

The component is `/components` folder:

https://github.com/mcastorena0316/Bookstore_CMS/blob/b833bc1e94a0d626112aa06353e5f49fefbbf10b/src/components/App.js#L7

File: src/components/Book.js (hunk 2 above)
Comment (Israel-Laguan) [reply src/components/Book.js]: Solved!

`eraseBook` function comes from `BooksList` using props drilling:

https://github.com/mcastorena0316/Bookstore_CMS/blob/b833bc1e94a0d626112aa06353e5f49fefbbf10b/src/components/Book.js#L11-L14

... so is not connected to `redux` anymore:

https://github.com/mcastorena0316/Bookstore_CMS/blob/b833bc1e94a0d626112aa06353e5f49fefbbf10b/src/components/Book.js#L50

File: src/containers/BookForm.js (hunk 4 above)
Comment (Israel-Laguan) [reply src/containers/BookForm.js]: Solved!

`console.log` is not anymore!:

https://github.com/mcastorena0316/Bookstore_CMS/blob/b833bc1e94a0d626112aa06353e5f49fefbbf10b/src/containers/BookForm.js#L36-L45

File: src/components/Book.js (hunk 3 above)
Comment (Israel-Laguan) [reply src/components/Book.js]: Solved!

We use book as prop:

https://github.com/mcastorena0316/Bookstore_CMS/blob/b833bc1e94a0d626112aa06353e5f49fefbbf10b/src/components/Book.js#L2-L14

**Pull request merged**